raimitigations/dataprocessing/imputer/iterative_imputer.py

```diff
@@ -0,0 +1,292 @@
+        self._set_dicts()
+
+    # -----------------------------------
+    def _set_dicts(self):
+        """
+        If the 'iterative_params' dictionary that specifies how to impute given data
+        is set to None, then create a default dictionary.
+        """
+        if self.iterative_params is None:
+            self.iterative_params = {
+                "estimator": BayesianRidge(),
+                "missing_values": np.nan,
+                "sample_posterior": False,
+                "max_iter": 10,
+                "tol": 1e-3,
+                "n_nearest_features": None,
+                "initial_strategy": "mean",
+                "imputation_order": "ascending",
+                "skip_complete": False,
+                "min_value": -np.inf,
+                "max_value": np.inf,
+                "random_state": None,
+            }
+
+    # -----------------------------------
+    def _check_valid_dict(self):
+        """
+        Checks if 'iterative_params' dictionary that specifis how to impute given data
+        is appropriately set.
+        """
+        param_err = type(self.iterative_params) != dict
+
+        if param_err:
+            raise ValueError("ERROR: 'iterative_params' is not a dict. Check the documentation for more information.")
+
+        keys = self.iterative_params.keys()
+        sklearn_params = [
+            "estimator",
+            "missing_values",
+            "sample_posterior",
+            "max_iter",
+            "tol",
+            "n_nearest_features",
+            "initial_strategy",
+            "imputation_order",
+            "skip_complete",
+            "min_value",
+            "max_value",
+            "random_state",
+        ]
+        for sklearn_param in sklearn_params:
+            if sklearn_param not in keys:
+                raise ValueError(
+                    "ERROR: expected the key "
+                    + f"{sklearn_param}"
+                    + " in the dictionary:\n"
+                    + f"iterative_params: {self.iterative_params}"
+                )
+
+    # -----------------------------------
+    def _fit(self):
+        """
+        Fit method complement used specifically for the current class.
+        The following steps are executed: (i) check if 'self.iterative_params'
+        dictionary is properly set, (ii) check for categorical columns
+        (iii) exclude categorical columns if 'enable_encoder' is false,
+        otherwise, apply ordinal encoder, (iv) create and fit the
+        IterativeImputer object over df.
+        """
+        self._check_valid_dict()
+
+        all_cat_cols = get_cat_cols(self.df)
+        all_num_cols = [value for value in list(self.df) if value not in all_cat_cols]
+
+        df_valid = pd.DataFrame()
+        self.valid_cols = []
+
+        if not isinstance(self.enable_encoder, bool):
+            raise ValueError("ERROR: 'enable_encoder' is a boolean parameter, use True/False.")
```

Comment:
Doesn't it make sense to do this check in the constructor method? This way, if there is any error with this parameter, the error will pop up in when creating the class, not when calling the fit() method.

Reply:
yes makes sense!